packages/swayze/lib/src/widgets/table_body/selections/fill_selections/fill_selection.dart

```diff
@@ -0,0 +1,270 @@
+      offset: leftTopPixelOffset,
+      border: getVisibleBorder(
+        range,
+        effectiveStyle.borderSide,
+      ),
+      duration: styleContext.selectionAnimationDuration,
+    );
+  }
+}
+
+/// An [ImplicitlyAnimatedWidget] that swiftly animates when [offset] and [size]
+/// changes.
+class _AnimatedFillSelection extends ImplicitlyAnimatedWidget {
+  final SelectionBorder border;
+  final Size size;
+  final Offset offset;
+
+  const _AnimatedFillSelection({
+    Key? key,
+    required this.border,
+    required this.size,
+    required this.offset,
+    required Duration duration,
+  }) : super(key: key, duration: duration);
+
+  @override
+  _AnimatedSelectionState createState() => _AnimatedSelectionState();
+}
+
+class _AnimatedSelectionState
+    extends AnimatedWidgetBaseState<_AnimatedFillSelection> {
+  late final viewportContext = ViewportContextProvider.of(context);
+
+  Tween<double>? _left;
+  Tween<double>? _top;
+  SizeTween? _size;
+
+  @override
+  void forEachTween(TweenVisitor<dynamic> visitor) {
+    _left = visitor(
+      _left,
+      widget.offset.dx,
+      (dynamic value) => Tween<double>(begin: value as double),
+    ) as Tween<double>?;
+
+    _top = visitor(
+      _top,
+      widget.offset.dy,
+      (dynamic value) => Tween<double>(begin: value as double),
+    ) as Tween<double>?;
+
+    _size = visitor(
+      _size,
+      widget.size,
+      (dynamic value) => SizeTween(begin: value as Size),
+    ) as SizeTween?;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final animation = this.animation;
+    final left = _left?.evaluate(animation) ?? 0;
+    final top = _top?.evaluate(animation) ?? 0;
+    final size = _size?.evaluate(animation) ?? Size.zero;
+
+    return FillSelectionPainter(
+      border: widget.border,
+      offset: Offset(left, top),
+      size: size,
+    );
+  }
+}
+
+/// A [LeafRenderObjectWidget] that render a fill selection.
+@visibleForTesting
+class FillSelectionPainter extends LeafRenderObjectWidget {
+  final SelectionBorder border;
+  final Offset offset;
+  final Size size;
+
+  const FillSelectionPainter({
+    required this.border,
+    required this.offset,
+    required this.size,
+  });
+
+  @override
+  _RenderFillSelectionPainter createRenderObject(BuildContext context) {
+    return _RenderFillSelectionPainter(
+      border,
+      offset,
+      size,
+    );
+  }
+
+  @override
+  void updateRenderObject(
+    BuildContext context,
+    _RenderFillSelectionPainter renderObject,
+  ) {
+    renderObject
+      ..border = border
+      ..offset = offset
+      ..definedSize = size;
+  }
+}
+
+class _RenderFillSelectionPainter extends RenderBox {
+  _RenderFillSelectionPainter(
+    this._border,
+    this._offset,
+    this._definedSize,
+  );
+
+  Offset _offset;
+  Offset get offset => _offset;
+  set offset(Offset value) {
+    if (_offset != value) {
+      _offset = value;
+      markNeedsPaint();
```

Comment:
doesnt need layout?

Reply:
Nope.

```
  @override
  void performLayout() {
    size = constraints.constrain(_definedSize);
  }
```

The layout only uses the given defined size. The offset is only used to translate the selection and paint it.